src/common/byteArrayHelper/include/byteArrayHelper.h

```diff
@@ -14,8 +14,15 @@
 
 #include <string>
 
+#ifdef __GNUC__
 #pragma GCC diagnostic push
 #pragma GCC diagnostic ignored "-Wunused-function"
+#endif
```

Comment:
As long as I understand, if -fgnuc-version=0 is not passed as a parameter to ClangApple, it will continue defining __GNUC__, which was its normal behaviour.

https://clang.llvm.org/docs/UsersManual.html#cmdoption-fgnuc-version

If you still think it is necessary, it could be changed in every place with __clang__, but the only advantage would be more clarity when reading it, not the behaviour.